DockerComposeContainer doesn't seem to work

Hi,
  I am trying to run DockerComposeContainer from a JUnit test, but it doesn't work. The application itself seems fine, as I can run it with no problem from command line:

` docker-compose up`

I set up   DockerComposeContainer  in my test with (in Kotlin):

```
companion object {

        class KDockerComposeContainer(path: File) : DockerComposeContainer<KDockerComposeContainer>(path)

        @ClassRule @JvmField
        val env = KDockerComposeContainer(File("../docker-compose.yml"))
    }
```

The test fails at initialization, with the following stack trace that does not really say much about what is wrong:
```
Connected to the target VM, address: '127.0.0.1:53858', transport: 'socket'
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
        ℹ︎ Checking the system...
        ✔ Docker version is newer than 1.6.0
        ✔ Docker environment has more than 2GB free
        ✔ File should be mountable
        ✔ Exposed port is accessible

Test ignored.

org.testcontainers.containers.ContainerLaunchException: Container startup failed

	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:189)
	at org.testcontainers.containers.ContainerisedDockerCompose.invoke(DockerComposeContainer.java:431)
	at org.testcontainers.containers.DockerComposeContainer.runWithCompose(DockerComposeContainer.java:151)
	at org.testcontainers.containers.DockerComposeContainer.pullImages(DockerComposeContainer.java:121)
	at org.testcontainers.containers.DockerComposeContainer.starting(DockerComposeContainer.java:108)
	at org.testcontainers.containers.FailureDetectingExternalResource$1.evaluate(FailureDetectingExternalResource.java:29)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:83)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:182)
	... 12 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:256)
	at org.testcontainers.containers.GenericContainer.lambda$start$0(GenericContainer.java:184)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:76)
	... 13 more
Caused by: java.lang.IllegalStateException: Container did not start correctly.
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:231)
	... 15 more
```

By using a debugger, I can see it fails to start `docker/compose:1.8.0`. Note: I am using Windows 10, with Docker version 17.06.0-ce-win18 (12627), and TestContainers 1.4.2. TestContainers work fines in the other tests in which I have single Docker images (i.e., no Compose).

Is this a bug, or am I missing something? (In this latter case, more info in the thrown exceptions would be useful)

@bsideup thx for the quick response, but we already using 1.4.2 and it didn't work for us on windows 10, thats the reason why i tried the version from the windows branch ...

> [INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.test.SampleE2EITCase
2017-09-28 20:11:30.392  INFO   --- [           main] tAndSystemPropertyClientProviderStrategy : Found docker client settings from environment
2017-09-28 20:11:30.442  INFO   --- [           main] o.t.d.DockerClientProviderStrategy       : Found Docker environment with Environment variables, system properties and defaults. Resolved: 
    dockerHost=tcp://127.0.0.1:2375
    apiVersion='{UNKNOWN_VERSION}'
    registryUrl='https://index.docker.io/v1/'
    registryUsername='user'
    registryPassword='null'
    registryEmail='null'
    dockerConfig='DefaultDockerClientConfig[dockerHost=tcp://127.0.0.1:2375,registryUsername=user,registryPassword=<null>,registryEmail=<null>,registryUrl=https://index.docker.io/v1/,dockerConfig=C:\Users\user/.docker,sslConfig=<null>,apiVersion=...]'
2017-09-28 20:11:30.452  INFO   --- [           main] org.testcontainers.DockerClientFactory   : Docker host IP address is 127.0.0.1
2017-09-28 20:11:30.801  INFO   --- [           main] org.testcontainers.DockerClientFactory   : Connected to docker: 
  Server Version: 17.06.2-ce
  API Version: 1.30
  Operating System: Alpine Linux v3.5
  Total Memory: 1987 MB
        ?? Checking the system...
        ? Docker version is newer than 1.6.0
        ? Docker environment has more than 2GB free
        ? File should be mountable
        ? Exposed port is accessible
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 9.512 s <<< FAILURE! - in org.test.SampleE2EITCase
[ERROR] org.test.SampleE2EITCase  Time elapsed: 9.511 s  <<< ERROR!
org.testcontainers.containers.ContainerLaunchException: Local Docker Compose not found. Is docker-compose on the PATH?

so it's doesn't find docker-compose command, even if you can execute it on command line

we have these docker and docker-compose version installed:

> D:\>docker --version
Docker version 17.06.2-ce, build cec0b72

> D:\>docker-compose --version
docker-compose version 1.14.0, build c7bdf9e3

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,7 +2,10 @@
 All notable changes to this project will be documented in this file.
 
 ## UNRELEASED
-_No changes yet_
+### Fixed
+- Fixed regression in 1.4.3 when using Docker Compose on Windows ([\#439](https://github.com/testcontainers/testcontainers-java/issues/439))
+
+### Changed
 
 ## [1.4.3] - 2017-10-14
 ### Fixed

--- a/core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java
+++ b/core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java
@@ -494,7 +494,7 @@ class LocalDockerCompose implements DockerCompose {
     @Override
     public void invoke() {
         // bail out early
-        if (!dockerComposeExecutableExists()) {
+        if (!CommandLine.executableExists(COMPOSE_EXECUTABLE)) {
             throw new ContainerLaunchException("Local Docker Compose not found. Is " + COMPOSE_EXECUTABLE + " on the PATH?");
         }
 
@@ -528,14 +528,6 @@ class LocalDockerCompose implements DockerCompose {
 
         } catch (Exception e) {
             throw new ContainerLaunchException("Error running local Docker Compose command: " + cmd, e);
-        }
-    }
-
-    private boolean dockerComposeExecutableExists() {
-        if (SystemUtils.IS_OS_WINDOWS) {
-            return CommandLine.executableExists(COMPOSE_EXECUTABLE + ".exe");
-        } else {
-            return CommandLine.executableExists(COMPOSE_EXECUTABLE);
         }
     }
 

